WordPress/src/test/java/org/wordpress/android/ui/sitecreation/MainVM/NewSiteCreationMainVMTest.kt

```diff
@@ -0,0 +1,209 @@
+    @Mock lateinit var wizardFinishedObserver: Observer<CreateSiteState>
+    @Mock lateinit var savedInstanceState: Bundle
+
+    @Captor
+    private val captorNavigationTarget: ArgumentCaptor<NavigationTarget>? = null
+
+    private lateinit var viewModel: NewSiteCreationMainVM
+
+    @Before
+    fun setUp() {
+        whenever(stepsProvider.getSteps()).thenReturn(SITE_CREATION_STEPS)
+        viewModel = NewSiteCreationMainVM(stepsProvider, tracker)
+        viewModel.start(null)
+        viewModel.navigationTargetObservable.observeForever(navigationTargetObserver)
+        viewModel.wizardFinishedObservable.observeForever(wizardFinishedObserver)
+    }
+
+    @Test
+    fun skipClickedResultsInNextStep() {
+        viewModel.onSkipClicked()
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun segmentSelectedResultsInNextStep() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun verticalSelectedResultsInNextStep() {
+        viewModel.onVerticalsScreenFinished(VERTICAL_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteInfoFinishedResultsInNextStep() {
+        viewModel.onInfoScreenFinished(SITE_TITLE, null)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun domainSelectedResultsInNextStep() {
+        viewModel.onDomainsScreenFinished(DOMAIN)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedSegment() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.segmentId).isEqualTo(SEGMENT_ID)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedVertical() {
+        viewModel.onVerticalsScreenFinished(VERTICAL_ID)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.verticalId).isEqualTo(VERTICAL_ID)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSiteInfo() {
+        viewModel.onInfoScreenFinished(SITE_TITLE, SITE_TAGLINE)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.siteTitle).isEqualTo(SITE_TITLE)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.siteTagLine).isEqualTo(SITE_TAGLINE)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedDomain() {
+        viewModel.onDomainsScreenFinished(DOMAIN)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.domain).isEqualTo(DOMAIN)
+    }
+
+    @Test
+    fun wizardFinishedInvokedOnSitePreviewCompleted() {
+        val state = SiteCreationCompleted(LOCAL_SITE_ID)
+        viewModel.onSitePreviewScreenFinished(state)
+
+        val captor = ArgumentCaptor.forClass(CreateSiteState::class.java)
+        verify(wizardFinishedObserver).onChanged(captor.capture())
+
+        assertThat(captor.value).isEqualTo(state)
+    }
+
+    @Test
+    fun nextStepAfterOnBackPressedIsCurrentStep() {
```

Comment:
I am a little bit worried about the setup of this test. The `onBackPressed` and `onSkipClicked` might not do a single thing and this test would still pass because it's tying two events together. I'd prefer if we checked that the `wizardManager.onBackPressed()` is called correctly when `viewModel.onBackPressed()` and not deal with the skip logic at all. What do you think? Is that even possible?